### PR TITLE
Helm: Added topologySpreadConstraints to read deployment component

### PR DESCRIPTION
**Pull request by sossickd**
**What this PR does / why we need it**:

Add missing topologySpreadConstraints for the read deployment component. Must have been missed in this [PR](https://github.com/grafana/loki/pull/11086)

Helm diff:

```console
helm diff upgrade loki ./loki -f ~/values-test.yaml -n logging --allow-unreleased
logging, loki-read, Deployment (apps) has changed:
  # Source: loki/templates/read/deployment-read.yaml
  apiVersion: apps/v1
  kind: Deployment
  .......
+       topologySpreadConstraints:
+         - labelSelector:
+             matchLabels:
+               app.kubernetes.io/component: read
+               app.kubernetes.io/instance: loki
+               app.kubernetes.io/name: loki
+           maxSkew: 1
+           nodeAffinityPolicy: Honor
+           nodeTaintsPolicy: Honor
+           topologyKey: topology.kubernetes.io/zone
+           whenUnsatisfiable: ScheduleAnyway
```

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by JStickler** [comment]
@sossickd could you please also update the helm version in `production/helm/loki/CHANGELOG.md`?

**Comment by MichelHollands** [comment]
@sossickd A new release was added so a new version is needed. Can you use 5.41.1? Once that is done I'll merge it.

**Comment by MichelHollands** [comment]
> @sossickd A new release was added so a new version is needed. Can you use 5.41.1? Once that is done I'll merge it.

@sossickd It was easy enough so I did it here. I'll merge as soon as the CI finishes.

File: production/helm/loki/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # loki
 
-![Version: 5.41.0](https://img.shields.io/badge/Version-5.41.0-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.9.3](https://img.shields.io/badge/AppVersion-2.9.3-informational?style=flat-square)
+![Version: 5.41.1](https://img.shields.io/badge/Version-5.41.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.9.2](https://img.shields.io/badge/AppVersion-2.9.2-informational?style=flat-square)
```
Comment (MichelHollands) [inline production/helm/loki/README.md:3]:
```suggestion
![Version: 5.41.1](https://img.shields.io/badge/Version-5.41.1-informational?style=flat-square) ![Type: application](https://img.shields.io/badge/Type-application-informational?style=flat-square) ![AppVersion: 2.9.3](https://img.shields.io/badge/AppVersion-2.9.3-informational?style=flat-square)
```

**Pull request merged**